tests/test_models.py

```diff
@@ -1,22 +1,188 @@
 # -*- coding: utf-8 -*-
+from django.conf import settings
+from django.core.exceptions import ValidationError
 from django.test import TestCase
 
-from djangocms_googlemap.models import GoogleMap
+from djangocms_googlemap.models import (
+    GoogleMap, GoogleMapMarker, GoogleMapRoute, get_templates,
+    ZOOM_LEVEL_CHOICES, MAP_TYPE_CHOICES, TRAVEL_MODE_CHOICES
+)
 
+from .helpers import get_filer_image
 
-class GoogleMapTestCase(TestCase):
+
+SAMPLE_STYLES = """
+[{"elementType": "geometry", "stylers": [{"color": "#242f3e"}]},
+{"elementType": "labels.text.stroke", "stylers": [{"color": "#242f3e"}]},
+{"elementType": "labels.text.fill", "stylers": [{"color": "#746855"}]}]
+"""
+
+
+class GoogleMapModelTestCase(TestCase):
 
     def setUp(self):
+        self.icon = get_filer_image("pin.png")
+
+    def tearDown(self):
+        if self.icon:
+            self.icon.delete()
+            del self.icon
+            with self.assertRaises(AttributeError):
+                print(self.icon)
+        GoogleMap.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMap.objects.all().count(), 0)
+        GoogleMapMarker.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMapMarker.objects.all().count(), 0)
+        GoogleMapRoute.objects.filter(pk=1).delete()
+        self.assertEqual(GoogleMapRoute.objects.all().count(), 0)
+
+    def test_settings(self):
+        self.assertEqual(get_templates(), [("default", "Default")])
+        settings.DJANGOCMS_GOOGLEMAP_TEMPLATES = [("feature", "Feature")]
+        self.assertEqual(get_templates(), [("default", "Default"), ("feature", "Feature")])
+
+    def test_googlemap_instance(self):
         GoogleMap.objects.create(
-            title='test',
+            template="default",
+            title="some title",
+            width="400px",
+            height="200px",
+            style=SAMPLE_STYLES,
+            lat=40.730610,
+            lng=-73.935242,
+            zoom=int(ZOOM_LEVEL_CHOICES[11][1]),
+            map_type_control=MAP_TYPE_CHOICES[0][1],
+            zoom_control=False,
+            street_view_control=False,
+            rotate_control=False,
+            scale_control=False,
+            fullscreen_control=False,
+            pan_control=False,
+            double_click_zoom=False,
+            draggable=False,
+            keyboard_shortcuts=False,
+            scrollwheel=False,
+        )
+        instance = GoogleMap.objects.all()
+        self.assertEqual(instance.count(), 1)
+        instance = GoogleMap.objects.get(pk=1)
```

Comment:
Hard-coding ID values is generally not the best idea; it's not guaranteed that they will always start from 1.

Reply:
In this test it should, cause its the first element I create

Reply:
Not necessarily – sequences do not get reset to their initial values when a transaction gets rolled back, which means if there was another test that also creates an instance, and if it were to run before this one, then this ID will be higher than 1. Also, if you run tests with `--keepdb` to speed them up, all sequences would have values left over from the previous run.

Sequences are only guaranteed to be reset if you use a `TransactionTestCase` with `reset_sequences` set to `True` (https://docs.djangoproject.com/en/2.2/topics/testing/advanced/#django.test.TransactionTestCase.reset_sequences), but that's not really something that you should do here.